### DISCUSSION TITLE
想请问哪些网段会影响PaoPaoDNS的功能实现？

### DISCUSSION THREAD

**Discussion by hsj1992** [kkkgo/PaoPaoDNS#88] [其他]
打算用bird2 OSPF协议做IP分流，弥补fakeip dns分流遗漏的如telegram这类直连IP的问题。
不知道这个操作行不行，就先试一下，另建了个fakeip tun模式 Mihomo旁路网关，用[nchnroutes](https://github.com/dndx/nchnroutes)生成非CNIP的routes4.conf，使用bird2的OSPF协议给ROS路由表，然后发现国内外的域名 nslookup 全变成fakeip进旁路网关了。
经测试把一些网段给注释掉后，国内域名就能正常解析了。
所以就有这个疑问。
```
...
#route 6.0.0.0/8 via "Meta";
#route 7.0.0.0/8 via "Meta";
#route 8.0.0.0/9 via "Meta";
route 8.192.0.0/10 via "Meta";
#route 9.0.0.0/8 via "Meta";
#route 11.0.0.0/8 via "Meta";
#route 12.0.0.0/8 via "Meta";
...
```

**Comment by kkkgo** [comment]
>需要注意的是，如果你的网络有“自动分流IP”的功能，请把容器的IP加入不分流的名单，因为权威DNS需要准确的IP去判断，IP分流会影响权威DNS的判断。

很难说你的结果全变成FAKEIP具体是什么情况（需要debug信息），如果是最新版本并且启用了USE_MARK_DATA功能，国内部分域名即使解析失败也不会出现FAKEIP，只会拒绝解析，但IP分流的确是会导致递归不准确。
直连IP的应用其实很有限，按需添加即可，取绝对补集一个问题是IP库并不一定准确，其次对实际路由拓扑侵入性太大。

**Reply by hsj1992** [reply]
在RouterOS v7里把PaoPaoDNS单独放一个表里，解决了受到静态路由表干扰的问题。

0、OSPF协议接收的路由表是在main路由表里。

1、添加一个bypass路由表。
```
/routing table add fib name=bypass
```

2、指定PaoPaoDNS本来要走main表的流量，去走bypass路由表。
```
/routing rule add src-address=192.168.31.4 routing-mark=main action=lookup table=bypass
```

3.添加静态路由让bypass表走pppoe-out1。
```
/ip route
add distance=1 dst-address=0.0.0.0/0 gateway=pppoe-out1 routing-table=bypass
```

4.设置内网流量，避免回环。
```
add distance=1 dst-address=10.0.0.0/8 gateway=bridge1 routing-table=bypass
add distance=1 dst-address=169.254.0.0/16 gateway=bridge1 routing-table=bypass
add distance=1 dst-address=172.16.0.0/12 gateway=bridge1 routing-table=bypass
add distance=1 dst-address=192.168.0.0/16 gateway=bridge1 routing-table=bypass
add distance=1 dst-address=224.0.0.0/4 gateway=bridge1 routing-table=bypass
add distance=1 dst-address=255.255.255.255/32 gateway=bridge1 routing-table=bypass
```

参考来自：[[Router OS]RouterOS用OSPF分流国内外IP，并放行PT流量，不使用mangle...](https://www.right.com.cn/forum/thread-8264237-1-1.html)

**Reply by kkkgo** [reply]
粗略看了下，这个拓扑要不走bt流量的话要么把bt软件装在网关上，要么单独划分一个网段给bt的设备用，可以说是跟dhcp option方案差不多的优缺点。

**Reply by hsj1992** [reply]
它这个帖子里是既装了bird2又装了BT软件和zerotier在pve上，所以要放行反而麻烦了。
如果我要用BT下载的话，我想可能就开个bt软件 docker容器，另设置IP和DNS，然后跟PaoPaoDNS一样同样放进bypass表里pppoe-out1直出，不知道效果会如何。
（说来说去又回到了ALL IN ONE BOOM）
在main表里享受IP分流和fakeip dns分流的设备，估计就开不来BT软件了。
划分网段的OSPF IP分流也有这个可以参考[基于路由协议的ip分流(RouterOS为例)](https://www.chiphell.com/thread-2438228-1-1.html)
但我是pve上跑的旁路网关+ROS CHR +PaoPaoDNS，就不按这个搞了。
